### PR TITLE
Format d bool support

### LINKED ISSUE #4939: {:d} format should support bool value.
{:d} format is supported to pass a boolean value. See the following test case. CPython can execute it and return expected result, while RustPython report a TypeError.

```
for i in range(12):
    print('{:d}, '.format(chr(i) in "ss"), end='')
    if i % 16 == 1:
        print()
```

Behaviors on CPython:
```
0, 0, 
0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 
```

Behaviors on RustPython
```
Traceback (most recent call last):
  File "/home/xxm/Desktop/RustPython/test.py", line 169, in <module>
    print('{:d}, '.format(chr(i) in "ss"), end='')
TypeError: unsupported format string passed to bool.__format__
```
Ubuntu 18.04
RustPython 0.2.0 [0284059](https://github.com/RustPython/RustPython/commit/02840593bc56ae416ba2646166628f1712d6cf43)
CPython 3.8.0, 3.9.0, 3.11.3

### PR DESCRIPTION
**Pull request by JazzGlobal**
Closes #4939

Probably more to do but im gonna let the action runner help me.

### CODE REVIEW

File: test.py
Comment (fanninpm) [inline test.py:1]: There might be a test in the CPython test suite that checks for this.

File: common/src/format.rs [hunk 1]
```diff
@@ -416,6 +416,18 @@ impl FormatSpec {
         }
     }
 
+    pub fn format_bool(&self, input: bool) -> Result<String, FormatSpecError> {
+        let x = match input {
+            true => 1,
+            false => 0,
+        };
```
Comment (fanninpm) [inline common/src/format.rs:423]: You could probably do something simpler like this:

```suggestion
        let x = u8::from(input);
```

(Just make sure Clippy is okay with it.)

File: vm/src/builtins/bool.rs [hunk 2]
```diff
@@ -110,12 +111,11 @@ impl Constructor for PyBool {
...
 impl PyBool {
     #[pymethod(magic)]
-    fn format(obj: PyObjectRef, format_spec: PyStrRef, vm: &VirtualMachine) -> PyResult<PyStrRef> {
-        if format_spec.is_empty() {
-            obj.str(vm)
-        } else {
-            Err(vm.new_type_error("unsupported format string passed to bool.__format__".to_owned()))
-        }
+    fn format(obj: PyObjectRef, spec: PyStrRef, vm: &VirtualMachine) -> PyResult<String> {
+        let new_bool = obj.try_to_bool(vm).unwrap();
```
Comment (fanninpm) [inline vm/src/builtins/bool.rs:115]: Does this work?
```suggestion
        let new_bool = obj.try_to_bool(vm)?;
```

Comment (JazzGlobal) [reply vm/src/builtins/bool.rs]: Yes, still not totally used to all the Rust shorthands :D

File: common/src/format.rs (hunk 1 above)
Comment (JazzGlobal) [reply common/src/format.rs]: Looks good, clippy didnt complain locally at least.

File: test.py
Comment (JazzGlobal) [reply test.py:1]: Speaking of tests, looks like I broke a handful. Executing `f'{1 != 2}'` throws an error.

**Review by youknowone — APPROVED** [review]
Thank you for contributing!

**Pull request merged**